diff --git a/typings/index.d.ts b/typings/index.d.ts
index 0224e4b22aafdaea3a0c8a7e50e0534776303099..f6780fe0869330fa21019c3de4edc9d17ca7bd7d 100644
--- a/typings/index.d.ts
+++ b/typings/index.d.ts
@@ -1,3 +1,5 @@
+import Util from '@ayako/bot/src/BaseClient/Bot/Util.js';
+import * as Sharding from 'discord-hybrid-sharding';
 import {
   ActionRowBuilder as BuilderActionRow,
   MessageActionRowComponentBuilder,
@@ -1047,6 +1049,8 @@ export class Client<Ready extends boolean = boolean> extends BaseClient {
     options?: { signal?: AbortSignal | undefined },
   ): AsyncIterableIterator<Emitter extends Client ? ClientEvents[Event] : any[]>;
 
+  public util: typeof Util;
+  public cluster?: Sharding.ClusterClient<typeof this>;
   public application: If<Ready, ClientApplication>;
   public channels: ChannelManager;
   public get emojis(): BaseGuildEmojiManager;
