diff --git a/typings/index.d.ts b/typings/index.d.ts
index c28cfb7f55ee07075153ca09bc91fdf2e6a7762d..917d1c2a1266598a59bb160f6a1bcdf0d3eb7823 100644
--- a/typings/index.d.ts
+++ b/typings/index.d.ts
@@ -1,3 +1,5 @@
+import Util from '@ayako/bot/src/BaseClient/Bot/Util.js';
+import * as Sharding from 'discord-hybrid-sharding';
 import {
   ActionRowBuilder as BuilderActionRow,
   MessageActionRowComponentBuilder,
@@ -1095,6 +1097,9 @@ export class Client<Ready extends boolean = boolean> extends BaseClient {
     options?: { signal?: AbortSignal | undefined },
   ): AsyncEventIterator<Emitter extends Client ? ClientEvents[Event] : any[]>;
 
+  public util: typeof Util;
+  public cluster?: Sharding.ClusterClient<typeof this>;
+
   public application: If<Ready, ClientApplication>;
   public channels: ChannelManager;
   public get emojis(): BaseGuildEmojiManager;
